python/ray/function_manager.py

```diff
@@ -30,6 +30,94 @@
                                    ["function", "function_name", "max_calls"])
 """FunctionExecutionInfo: A named tuple storing remote function information."""
 
+# Avoid circle import of worker.py.
+NIL_FUNCTION_ID = ray_constants.NIL_JOB_ID
+
+
+class FunctionDescriptor(object):
+    """A class used to describe a python function.
+
+    Attributes:
+        _module_name: The module name of this function.
```

Comment:
Since these are private attributes but are exposed as properties, they should probably be documented as properties: 

> Properties created with the ``@property`` decorator should be documented
    in the property's getter method.

https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html